spark/v3.1/spark/src/jmh/java/org/apache/iceberg/spark/source/WritersBenchmark.java

```diff
@@ -79,6 +79,7 @@ public abstract class WritersBenchmark extends IcebergSourceBenchmark {
   private PartitionSpec unpartitionedSpec;
   private PartitionSpec partitionedSpec;
 
+  @Override
   protected abstract FileFormat fileFormat();
```

Comment:
What about just removing this method? We already have one in the parent class.

Reply:
We can probably do that for all Spark versions. It is a bit awkward to have an abstract method that overrides a method in the parent class.

Reply:
It is odd to declare an Override of an abstract method and keep it abstract.

This matches the same file in Spark 3.2 code, so I prefer this for consistency reasons.

But I don’t have a strong preference. I could update this one and then leave the other versions alone one or update all of them to remove it (since it’s in the parent class).

Reply:
Ok. I’ll try removing all of them as it is very strange to have an override for an abstract method and still be abstract.

Reply:
Updated to remove the redeclaration instead. Thanks @aokolnychyi!

Reply:
Trying that, I got errors running it (see the comment on the main thread).

So I've added back in the overridden abstract method.

Reply:
For spark versions 3.0 and lower, the function is _not_ declared in the base class `IcebergSourceBenchmark`. So I added it there so that all of the versions would be consistent.

I don't _love_ this approach, and I'm happy to try removing it from there and simply making it part of the `WritersBenchmark` instead. But at least this is consistent and works.